source/simulator/assemblers/stokes.cc

```diff
@@ -124,8 +126,8 @@ namespace aspect
           const double one_over_eta = 1. / eta;
 
           const double JxW = scratch.finite_element_values.JxW(q);
-
-          if (this->get_parameters().use_full_A_block_preconditioner == false)
+//assembles stokes preconditioner ? qxh
+          if (this->get_parameters().use_full_A_block_preconditioner == false && !this->get_parameters().use_bfbt) //not full a and not bfbt
             {
```

Comment:
We need to split up the assembly code into something sensible. I think we could split the computation of the velocity and the pressure block (at least within this assembler).

Reply:
@gassmoeller ?